.yalc/msupply-ble-service/src/Bluetooth/BleService.ts

```diff
@@ -79,9 +79,20 @@ export class BleService {
   connectAndDiscoverServices = async (deviceDescriptor: string): Promise<TypedDevice> => {
     this.logger.info('connectAndDiscoverServices', { deviceDescriptor });
     const device = this.utils.deviceDescriptorToDevice(deviceDescriptor);
-    const deviceIsConnected = await this.manager.isDeviceConnected(device.id);
-    this.logger.info('deviceIsConnected?', { deviceIsConnected });
-    if (deviceIsConnected) {
+    // the Blue Maestro devices are incorrectly reporting connection status
+    // thus: deviceIsConnected?	{ deviceIsConnected: true }
+    // then if disconnecting [BleError: Device D7:D6:67:E0:02:34 is not connected]
+    // in which case an error is thrown when trying to connect: [BleError: Device ? is already connected]
+    // to work around this, we disconnect the device, ignoring any errors, before connecting again
+    if (device.deviceType === BLUE_MAESTRO) {
+      try {
+        await this.manager.cancelDeviceConnection(device.id);
+      } catch {
+        // ignore error
+      }
+    } else {
+      const deviceIsConnected = await this.manager.isDeviceConnected(device.id);
+      this.logger.info('deviceIsConnected?', { deviceIsConnected });
       await this.manager.cancelDeviceConnection(device.id);
```

Comment:
we call `connectAndDiscoverServices` at the start of each of the commands.. I wonder if also cancelling the connection after each command, also, would somehow make things more stable 😅

Reply:
good point - I half wondered about that. will have a try!